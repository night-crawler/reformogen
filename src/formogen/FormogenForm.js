import { connect } from 'react-redux';
import { createStructuredSelector } from 'reselect';

import { FormogenFormComponent } from './FormogenFormComponent';
import { 
  
} from './selectors';


function mapDispatchToProps(dispatch) {
  return {
    dispatch,
  };
}

function mergeProps(stateProps, dispatchProps, ownProps) {
  const { dispatch } = dispatchProps;

  return {
    ...ownProps,
    ...stateProps,
    ...dispatchProps,

    dispatch: undefined,
  };
}

export const FormogenForm = connect(
  createStructuredSelector({

  }),
  mapDispatchToProps,
  mergeProps,
)(FormogenFormComponent);
