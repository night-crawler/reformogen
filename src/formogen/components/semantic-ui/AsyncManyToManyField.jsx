import React from 'react';
import { Form } from 'semantic-ui-react';
import Select from 'react-select';
import _ from 'lodash';
import 'react-select/dist/react-select.css';
import fetch from 'isomorphic-fetch';
import Label from './Label';
import propTypes from '../fieldPropTypes';
import * as URI from 'urijs';

import ModelInstanceOption from './ReactSelectOptionComponent';
import ModelInstanceValue from './ReactSelectValueComponent';


AsyncManyToManyField.propTypes = propTypes;
export default function AsyncManyToManyField(props) {
    /*
        "name": "inspire_source",
        "verbose_name": "inspire source",
        "help_text": "",
        "blank": true,
        "null": false,
        "editable": true,
        "type": "ForeignKeyField",
        "required": false,
        "data": []
    */
    const handleChange = (val) => {
        let plainIds = val.map( ({ id }) => (id*1) );
        props.onChange(null, {name: props.name, value: plainIds});
    };

    const loadOptions = (input, callback) => {
        let uri = URI(props.data).addSearch({q: input});

        fetch(uri)
            .then(response => response.json())
            .then(json => {
                if (_.isArray(json)) {  /* without pagination */
                    callback(null, { complete: false, options: json, });
                } else {  /* with pagination */
                    callback(null, { complete: false, options: json.results, });
                }
            });
    };

    let _props = {
        clearable: !props.required,
        closeOnSelect: true,
        disabled: !props.editable,
        multi: true,
        onChange: handleChange,
        placeholder: props.placeholder,

        value: props.value,
        inputProps: {type: 'react-type'},  // fixes broken semantic markup
        removeSelected: true,

        valueKey: 'id',  /* server-side model should provide `id` for object */
        // simpleValue: true,  /* we have no need to store whole objects in parent's state */

        optionComponent: ModelInstanceOption,
        valueComponent: ModelInstanceValue,

        // https://github.com/JedWatson/react-select#advanced-filters
        // we use server-side filters, Select should not filter it again
        filterOption: () => true,
        loadOptions: loadOptions,

        // rtl: this.state.rtl,
    };

    if (_.isFunction(props.updateProps)) {
        _props = props.updateProps(_props, props);
    }

    return (
        <Form.Field required={ props.required } disabled={ !props.editable }>
            <Label { ...props } />
            <Select.Async { ..._props } />
            { !props.helpTextOnHover ? <span className="help-text">{ props.help_text }</span> : ''}
        </Form.Field>
    );
}
