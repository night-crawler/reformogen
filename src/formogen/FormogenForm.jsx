import React, { Component } from 'react';
import PropTypes from 'prop-types';

import * as suiFields from './semantic-ui/fields';


const defaultFieldComponentMap = {
  CharField: suiFields.CharField,
  TextField: suiFields.TextField,

  DateTimeField: suiFields.DateTimeField,
  DateField: suiFields.DateField,
  TimeField: suiFields.TimeField,

  PositiveSmallIntegerField: suiFields.IntegerField,
  SmallIntegerField: suiFields.IntegerField,
  IntegerField: suiFields.IntegerField,
  PositiveIntegerField: suiFields.IntegerField,
  DecimalField: suiFields.IntegerField,
  FloatField: suiFields.IntegerField,

  // FileField: suiFields.DropzoneField,
  BooleanField: suiFields.BooleanField,

  ChoiceField: suiFields.AutocompleteChoiceField,

  GenericField: suiFields.GenericField,
};

export class FormogenForm extends Component {
  static propTypes = {
    getFormId: PropTypes.func,

    loading: PropTypes.bool,
    locale: PropTypes.string,
    showHeader: PropTypes.bool,
    upperFirstLabels: PropTypes.bool,
    helpTextOnHover: PropTypes.bool,

    /** a mapping of FieldType to FieldComponent: { BooleanField: <BooleanFieldClass> }  */
    componentMap: PropTypes.object,

    /** form description */
    metaData: PropTypes.arrayOf(PropTypes.object).isRequired,

    /* formdata */
    formData: PropTypes.object,

    actions: PropTypes.shape({
      bootstrap: PropTypes.func,
    }),
  };
  static defaultProps = {
    componentMap: defaultFieldComponentMap,
    getFormId: props => {}
  };

  render() {
    return (
      <div>
        
      </div>
    );
  }

  componentDidMount = () => {
    this.props.actions.bootstrap();
  }
}
