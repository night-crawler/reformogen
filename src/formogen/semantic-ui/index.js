// fields import
import AsyncForeignKeyField from './fields/AsyncForeignKeyField';
import AsyncManyToManyField from './fields/AsyncManyToManyField';
import ManyToManyField from './fields/ManyToManyField';
import ForeignKeyField from './fields/InlineForeignKeyField';
import AutocompleteChoiceField from './fields/AutocompleteChoiceField';
import GenericField from './fields/GenericField';
import IntegerField from './fields/IntegerField';
import BooleanField from './fields/BooleanField';
import CharField from './fields/CharField';
import TextField from './fields/TextField';
import DateTimeField from './fields/DateTimeField';
import DateField from './fields/DateField';
import TimeField from './fields/TimeField';
import DropzoneField from './fields/DropzoneField';
// common import
import Label from './common/Label';
import CaptionTruncator from './common/CaptionTruncator';
import Form from './common/Form';
import ModalForm from './common/ModalForm';
import { ErrorsList } from './common/ErrorsList';
import NonFieldErrorsList from './common/NonFieldErrorsList';

import 'react-datepicker/dist/react-datepicker.css';
import 'react-times/css/material/default.css';

import './index.css';

export const fields = {
  AutocompleteChoiceField,

  ForeignKeyField,
  AsyncForeignKeyField,

  ManyToManyField,
  AsyncManyToManyField,

  IntegerField,
  BooleanField,
  CharField,
  TextField,

  DateTimeField,
  DateField,
  TimeField,

  DropzoneField,

  GenericField,
};

export const common = {
  Label,
  CaptionTruncator,

  Form,
  ModalForm,

  ErrorsList,
  NonFieldErrorsList,
};