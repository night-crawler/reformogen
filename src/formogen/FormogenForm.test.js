import { FormogenForm } from './FormogenForm';

import * as sui from '~/formogen/semantic-ui';

export const fullPossibleMetadata = {
  'title': 'Create AllModelFields',
  'description': '',
  'fields': [
    {
      'name': 'f_date',
      'verbose_name': 'date',
      'help_text': '',
      'blank': false,
      'null': false,
      'editable': true,
      'type': 'DateField',
      'required': true,
      'default': '2018-11-06'
    },
    {
      'name': 'f_time',
      'verbose_name': 'time',
      'help_text': '',
      'blank': false,
      'null': false,
      'editable': true,
      'type': 'TimeField',
      'required': true,
      'default': '08:30:35.793611'
    },
    {
      'name': 'f_dt',
      'verbose_name': 'date and time',
      'help_text': '',
      'blank': false,
      'null': false,
      'editable': true,
      'type': 'DateTimeField',
      'required': true,
      'default': '2018-11-06T08:30:35.793628Z'
    },
    {
      'name': 'f_integer',
      'verbose_name': 'integer',
      'help_text': '',
      'blank': false,
      'null': false,
      'editable': true,
      'type': 'IntegerField',
      'required': true,
      'default': -22
    },
    {
      'name': 'f_positive_integer',
      'verbose_name': 'positive integer',
      'help_text': '',
      'blank': false,
      'null': false,
      'editable': true,
      'type': 'PositiveIntegerField',
      'required': true,
      'default': 0
    },
    {
      'name': 'f_small_integer',
      'verbose_name': 'small integer',
      'help_text': '',
      'blank': false,
      'null': false,
      'editable': true,
      'type': 'SmallIntegerField',
      'required': true,
      'default': 13
    },
    {
      'name': 'f_positive_small_integer',
      'verbose_name': 'positive small integer',
      'help_text': '',
      'blank': false,
      'null': false,
      'editable': true,
      'type': 'PositiveSmallIntegerField',
      'required': true,
      'default': 222
    },
    {
      'name': 'f_decimal',
      'verbose_name': 'decimal',
      'help_text': '',
      'blank': false,
      'null': false,
      'editable': true,
      'decimal_places': 4,
      'max_digits': 10,
      'type': 'DecimalField',
      'required': true,
      'default': 3.1415
    },
    {
      'name': 'f_float',
      'verbose_name': 'float',
      'help_text': '',
      'blank': false,
      'null': false,
      'editable': true,
      'type': 'FloatField',
      'required': true,
      'default': 3.14
    },
    {
      'name': 'f_fk_embed',
      'verbose_name': 'embedded foreign key',
      'help_text': '',
      'blank': true,
      'null': true,
      'editable': true,
      'type': 'ForeignKey',
      'required': false,
      'data': [
        {
          'id': 1,
          'name': 'Author Вася'
        },
        {
          'id': 2,
          'name': 'Author asd Вася'
        }
      ]
    },
    {
      'name': 'f_fk_rel',
      'verbose_name': 'foreign key with relation',
      'help_text': '',
      'blank': true,
      'null': true,
      'editable': true,
      'type': 'ForeignKey',
      'required': false,
      'data': 'http://localhost:8000/api/v1/sample/books/'
    },
    {
      'name': 'f_choice',
      'verbose_name': 'choice',
      'help_text': '',
      'blank': false,
      'null': false,
      'editable': true,
      'type': 'PositiveSmallIntegerField',
      'required': true,
      'default': 255,
      'choices': [
        [
          0,
          'dead'
        ],
        [
          255,
          'alive'
        ],
        [
          30,
          'dried'
        ]
      ]
    },
    {
      'name': 'f_file',
      'verbose_name': 'short preview sample',
      'help_text': '',
      'blank': false,
      'null': false,
      'editable': true,
      'max_length': 100,
      'type': 'FileField',
      'required': true,
      'upload_url': 'http://localhost:8000/api/v1/sample/all/accept_file/'
    },
    {
      'name': 'f_photo',
      'verbose_name': 'photo',
      'help_text': 'logo image',
      'blank': true,
      'null': true,
      'editable': true,
      'max_length': 100,
      'type': 'FileField',
      'required': false,
      'upload_url': 'http://localhost:8000/api/v1/sample/all/accept_file/'
    },
    {
      'name': 'f_char_field',
      'verbose_name': 'char field',
      'help_text': 'charfield',
      'blank': false,
      'null': false,
      'editable': true,
      'max_length': 255,
      'type': 'CharField',
      'required': true
    },
    {
      'name': 'f_text_field',
      'verbose_name': 'text field',
      'help_text': '',
      'blank': true,
      'null': false,
      'editable': true,
      'type': 'TextField',
      'required': false
    },
    {
      'name': 'f_m2m_embed',
      'verbose_name': 'embedded M2M',
      'help_text': '',
      'blank': true,
      'null': false,
      'editable': true,
      'type': 'ManyToManyField',
      'required': false,
      'data': [
        {
          'id': 1,
          'name': 'Author Вася'
        },
        {
          'id': 2,
          'name': 'Author asd Вася'
        }
      ]
    },
    {
      'name': 'f_m2m_rel',
      'verbose_name': 'related M2M',
      'help_text': '',
      'blank': true,
      'null': false,
      'editable': true,
      'type': 'ManyToManyField',
      'required': false,
      'data': 'http://localhost:8000/api/v1/sample/authors/'
    }
  ]
};

it('<FormogenForm />', () => {
  const wrapper = shallow(
    <FormogenForm
      formId='bla'
      metaData={ fullPossibleMetadata }  
      getFieldComponent={ sui.getFieldComponentForType }
      getFormComponent={ () => sui.FormComponent }

      actions={ {
        bootstrap: () => {}
      } }
    />
  );
  expect(wrapper).toMatchSnapshot();
});
