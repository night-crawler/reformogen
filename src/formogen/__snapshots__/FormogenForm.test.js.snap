// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FormogenForm /> 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <FormogenForm
    actions={
      Object {
        "bootstrap": [Function],
      }
    }
    errorsFieldMap={Object {}}
    fieldsets={
      Array [
        Object {
          "fields": Array [
            "*",
          ],
          "getDisplayOptions": [Function],
          "header": null,
        },
      ]
    }
    formId="bla"
    getFieldComponent={[Function]}
    getFormComponent={[Function]}
    isTitleVisible={true}
    metaData={
      Object {
        "description": "",
        "fields": Array [
          Object {
            "blank": false,
            "default": "2018-11-06",
            "editable": true,
            "help_text": "",
            "name": "f_date",
            "null": false,
            "required": true,
            "type": "DateField",
            "verbose_name": "date",
          },
          Object {
            "blank": false,
            "default": "08:30:35.793611",
            "editable": true,
            "help_text": "",
            "name": "f_time",
            "null": false,
            "required": true,
            "type": "TimeField",
            "verbose_name": "time",
          },
          Object {
            "blank": false,
            "default": "2018-11-06T08:30:35.793628Z",
            "editable": true,
            "help_text": "",
            "name": "f_dt",
            "null": false,
            "required": true,
            "type": "DateTimeField",
            "verbose_name": "date and time",
          },
          Object {
            "blank": false,
            "default": -22,
            "editable": true,
            "help_text": "",
            "name": "f_integer",
            "null": false,
            "required": true,
            "type": "IntegerField",
            "verbose_name": "integer",
          },
          Object {
            "blank": false,
            "default": 0,
            "editable": true,
            "help_text": "",
            "name": "f_positive_integer",
            "null": false,
            "required": true,
            "type": "PositiveIntegerField",
            "verbose_name": "positive integer",
          },
          Object {
            "blank": false,
            "default": 13,
            "editable": true,
            "help_text": "",
            "name": "f_small_integer",
            "null": false,
            "required": true,
            "type": "SmallIntegerField",
            "verbose_name": "small integer",
          },
          Object {
            "blank": false,
            "default": 222,
            "editable": true,
            "help_text": "",
            "name": "f_positive_small_integer",
            "null": false,
            "required": true,
            "type": "PositiveSmallIntegerField",
            "verbose_name": "positive small integer",
          },
          Object {
            "blank": false,
            "decimal_places": 4,
            "default": 3.1415,
            "editable": true,
            "help_text": "",
            "max_digits": 10,
            "name": "f_decimal",
            "null": false,
            "required": true,
            "type": "DecimalField",
            "verbose_name": "decimal",
          },
          Object {
            "blank": false,
            "default": 3.14,
            "editable": true,
            "help_text": "",
            "name": "f_float",
            "null": false,
            "required": true,
            "type": "FloatField",
            "verbose_name": "float",
          },
          Object {
            "blank": true,
            "data": Array [
              Object {
                "id": 1,
                "name": "Author Вася",
              },
              Object {
                "id": 2,
                "name": "Author asd Вася",
              },
            ],
            "editable": true,
            "help_text": "",
            "name": "f_fk_embed",
            "null": true,
            "required": false,
            "type": "ForeignKey",
            "verbose_name": "embedded foreign key",
          },
          Object {
            "blank": true,
            "data": "http://localhost:8000/api/v1/sample/books/",
            "editable": true,
            "help_text": "",
            "name": "f_fk_rel",
            "null": true,
            "required": false,
            "type": "ForeignKey",
            "verbose_name": "foreign key with relation",
          },
          Object {
            "blank": false,
            "choices": Array [
              Array [
                0,
                "dead",
              ],
              Array [
                255,
                "alive",
              ],
              Array [
                30,
                "dried",
              ],
            ],
            "default": 255,
            "editable": true,
            "help_text": "",
            "name": "f_choice",
            "null": false,
            "required": true,
            "type": "PositiveSmallIntegerField",
            "verbose_name": "choice",
          },
          Object {
            "blank": false,
            "editable": true,
            "help_text": "",
            "max_length": 100,
            "name": "f_file",
            "null": false,
            "required": true,
            "type": "FileField",
            "upload_url": "http://localhost:8000/api/v1/sample/all/accept_file/",
            "verbose_name": "short preview sample",
          },
          Object {
            "blank": true,
            "editable": true,
            "help_text": "logo image",
            "max_length": 100,
            "name": "f_photo",
            "null": true,
            "required": false,
            "type": "FileField",
            "upload_url": "http://localhost:8000/api/v1/sample/all/accept_file/",
            "verbose_name": "photo",
          },
          Object {
            "blank": false,
            "editable": true,
            "help_text": "charfield",
            "max_length": 255,
            "name": "f_char_field",
            "null": false,
            "required": true,
            "type": "CharField",
            "verbose_name": "char field",
          },
          Object {
            "blank": true,
            "editable": true,
            "help_text": "",
            "name": "f_text_field",
            "null": false,
            "required": false,
            "type": "TextField",
            "verbose_name": "text field",
          },
          Object {
            "blank": true,
            "data": Array [
              Object {
                "id": 1,
                "name": "Author Вася",
              },
              Object {
                "id": 2,
                "name": "Author asd Вася",
              },
            ],
            "editable": true,
            "help_text": "",
            "name": "f_m2m_embed",
            "null": false,
            "required": false,
            "type": "ManyToManyField",
            "verbose_name": "embedded M2M",
          },
          Object {
            "blank": true,
            "data": "http://localhost:8000/api/v1/sample/authors/",
            "editable": true,
            "help_text": "",
            "name": "f_m2m_rel",
            "null": false,
            "required": false,
            "type": "ManyToManyField",
            "verbose_name": "related M2M",
          },
        ],
        "title": "Create AllModelFields",
      }
    }
    upperFirstLabels={true}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": "FormComponent-bla",
    "nodeType": "function",
    "props": Object {
      "fieldsets": Array [
        Object {
          "fields": Array [
            <DateField
              blank={false}
              dateFormat="YYYY.MM.DD"
              default="2018-11-06"
              displayOptions={
                Object {
                  "width": 16,
                }
              }
              dropdownMode="select"
              editable={true}
              formId="bla"
              help_text=""
              name="f_date"
              null={false}
              required={true}
              showMonthDropdown={true}
              showTimeSelect={false}
              showYearDropdown={true}
              todayButton="Now"
              type="DateField"
              upperFirstLabel={true}
              verbose_name="date"
            />,
            <TimeField
              blank={false}
              default="08:30:35.793611"
              displayOptions={
                Object {
                  "width": 16,
                }
              }
              editable={true}
              formId="bla"
              help_text=""
              name="f_time"
              null={false}
              required={true}
              type="TimeField"
              upperFirstLabel={true}
              verbose_name="time"
            />,
            <DateTimeField
              blank={false}
              dateFormat="YYYY.MM.DD HH:mm"
              default="2018-11-06T08:30:35.793628Z"
              displayOptions={
                Object {
                  "width": 16,
                }
              }
              dropdownMode="select"
              editable={true}
              formId="bla"
              help_text=""
              name="f_dt"
              null={false}
              required={true}
              timeFormat="HH:mm"
              timeIntervals={30}
              todayButton="Now"
              type="DateTimeField"
              upperFirstLabel={true}
              verbose_name="date and time"
            />,
            <IntegerField
              blank={false}
              default={-22}
              displayOptions={
                Object {
                  "width": 16,
                }
              }
              editable={true}
              formId="bla"
              help_text=""
              name="f_integer"
              null={false}
              required={true}
              type="IntegerField"
              upperFirstLabel={true}
              verbose_name="integer"
            />,
            <IntegerField
              blank={false}
              default={0}
              displayOptions={
                Object {
                  "width": 16,
                }
              }
              editable={true}
              formId="bla"
              help_text=""
              name="f_positive_integer"
              null={false}
              required={true}
              type="PositiveIntegerField"
              upperFirstLabel={true}
              verbose_name="positive integer"
            />,
            <IntegerField
              blank={false}
              default={13}
              displayOptions={
                Object {
                  "width": 16,
                }
              }
              editable={true}
              formId="bla"
              help_text=""
              name="f_small_integer"
              null={false}
              required={true}
              type="SmallIntegerField"
              upperFirstLabel={true}
              verbose_name="small integer"
            />,
            <IntegerField
              blank={false}
              default={222}
              displayOptions={
                Object {
                  "width": 16,
                }
              }
              editable={true}
              formId="bla"
              help_text=""
              name="f_positive_small_integer"
              null={false}
              required={true}
              type="PositiveSmallIntegerField"
              upperFirstLabel={true}
              verbose_name="positive small integer"
            />,
            <IntegerField
              blank={false}
              decimal_places={4}
              default={3.1415}
              displayOptions={
                Object {
                  "width": 16,
                }
              }
              editable={true}
              formId="bla"
              help_text=""
              max_digits={10}
              name="f_decimal"
              null={false}
              required={true}
              type="DecimalField"
              upperFirstLabel={true}
              verbose_name="decimal"
            />,
            <IntegerField
              blank={false}
              default={3.14}
              displayOptions={
                Object {
                  "width": 16,
                }
              }
              editable={true}
              formId="bla"
              help_text=""
              name="f_float"
              null={false}
              required={true}
              type="FloatField"
              upperFirstLabel={true}
              verbose_name="float"
            />,
            <InlineForeignKeyField
              blank={true}
              data={
                Array [
                  Object {
                    "id": 1,
                    "name": "Author Вася",
                  },
                  Object {
                    "id": 2,
                    "name": "Author asd Вася",
                  },
                ]
              }
              displayOptions={
                Object {
                  "width": 16,
                }
              }
              editable={true}
              formId="bla"
              getOptionLabel={[Function]}
              getOptionValue={[Function]}
              help_text=""
              isRtl={false}
              name="f_fk_embed"
              null={true}
              required={false}
              type="ForeignKey"
              upperFirstLabel={true}
              verbose_name="embedded foreign key"
            />,
            <GenericField
              blank={true}
              data="http://localhost:8000/api/v1/sample/books/"
              displayOptions={
                Object {
                  "width": 16,
                }
              }
              editable={true}
              formId="bla"
              help_text=""
              name="f_fk_rel"
              null={true}
              required={false}
              type="ForeignKey"
              upperFirstLabel={true}
              verbose_name="foreign key with relation"
            />,
            <AutocompleteChoiceField
              blank={false}
              choices={
                Array [
                  Array [
                    0,
                    "dead",
                  ],
                  Array [
                    255,
                    "alive",
                  ],
                  Array [
                    30,
                    "dried",
                  ],
                ]
              }
              default={255}
              displayOptions={
                Object {
                  "width": 16,
                }
              }
              editable={true}
              formId="bla"
              getOptionLabel={[Function]}
              getOptionValue={[Function]}
              help_text=""
              name="f_choice"
              null={false}
              required={true}
              type="PositiveSmallIntegerField"
              upperFirstLabel={true}
              verbose_name="choice"
            />,
            <GenericField
              blank={false}
              displayOptions={
                Object {
                  "width": 16,
                }
              }
              editable={true}
              formId="bla"
              help_text=""
              max_length={100}
              name="f_file"
              null={false}
              required={true}
              type="FileField"
              upload_url="http://localhost:8000/api/v1/sample/all/accept_file/"
              upperFirstLabel={true}
              verbose_name="short preview sample"
            />,
            <GenericField
              blank={true}
              displayOptions={
                Object {
                  "width": 16,
                }
              }
              editable={true}
              formId="bla"
              help_text="logo image"
              max_length={100}
              name="f_photo"
              null={true}
              required={false}
              type="FileField"
              upload_url="http://localhost:8000/api/v1/sample/all/accept_file/"
              upperFirstLabel={true}
              verbose_name="photo"
            />,
            <CharField
              blank={false}
              displayOptions={
                Object {
                  "width": 16,
                }
              }
              editable={true}
              formId="bla"
              help_text="charfield"
              max_length={255}
              name="f_char_field"
              null={false}
              required={true}
              type="CharField"
              upperFirstLabel={true}
              verbose_name="char field"
            />,
            <TextField
              blank={true}
              displayOptions={
                Object {
                  "width": 16,
                }
              }
              editable={true}
              formId="bla"
              help_text=""
              name="f_text_field"
              null={false}
              required={false}
              type="TextField"
              upperFirstLabel={true}
              verbose_name="text field"
            />,
            <InlineManyToManyField
              blank={true}
              closeOnSelect={true}
              data={
                Array [
                  Object {
                    "id": 1,
                    "name": "Author Вася",
                  },
                  Object {
                    "id": 2,
                    "name": "Author asd Вася",
                  },
                ]
              }
              displayOptions={
                Object {
                  "width": 16,
                }
              }
              editable={true}
              formId="bla"
              getOptionLabel={[Function]}
              getOptionValue={[Function]}
              help_text=""
              name="f_m2m_embed"
              null={false}
              required={false}
              type="ManyToManyField"
              upperFirstLabel={true}
              value={Array []}
              verbose_name="embedded M2M"
            />,
            <GenericField
              blank={true}
              data="http://localhost:8000/api/v1/sample/authors/"
              displayOptions={
                Object {
                  "width": 16,
                }
              }
              editable={true}
              formId="bla"
              help_text=""
              name="f_m2m_rel"
              null={false}
              required={false}
              type="ManyToManyField"
              upperFirstLabel={true}
              verbose_name="related M2M"
            />,
          ],
          "header": null,
        },
      ],
      "formId": undefined,
      "isTitleVisible": true,
      "loading": false,
      "nonFieldErrorsMap": undefined,
      "onSubmit": [Function],
      "submitComponent": [Function],
      "title": "Create AllModelFields",
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": "FormComponent-bla",
      "nodeType": "function",
      "props": Object {
        "fieldsets": Array [
          Object {
            "fields": Array [
              <DateField
                blank={false}
                dateFormat="YYYY.MM.DD"
                default="2018-11-06"
                displayOptions={
                  Object {
                    "width": 16,
                  }
                }
                dropdownMode="select"
                editable={true}
                formId="bla"
                help_text=""
                name="f_date"
                null={false}
                required={true}
                showMonthDropdown={true}
                showTimeSelect={false}
                showYearDropdown={true}
                todayButton="Now"
                type="DateField"
                upperFirstLabel={true}
                verbose_name="date"
              />,
              <TimeField
                blank={false}
                default="08:30:35.793611"
                displayOptions={
                  Object {
                    "width": 16,
                  }
                }
                editable={true}
                formId="bla"
                help_text=""
                name="f_time"
                null={false}
                required={true}
                type="TimeField"
                upperFirstLabel={true}
                verbose_name="time"
              />,
              <DateTimeField
                blank={false}
                dateFormat="YYYY.MM.DD HH:mm"
                default="2018-11-06T08:30:35.793628Z"
                displayOptions={
                  Object {
                    "width": 16,
                  }
                }
                dropdownMode="select"
                editable={true}
                formId="bla"
                help_text=""
                name="f_dt"
                null={false}
                required={true}
                timeFormat="HH:mm"
                timeIntervals={30}
                todayButton="Now"
                type="DateTimeField"
                upperFirstLabel={true}
                verbose_name="date and time"
              />,
              <IntegerField
                blank={false}
                default={-22}
                displayOptions={
                  Object {
                    "width": 16,
                  }
                }
                editable={true}
                formId="bla"
                help_text=""
                name="f_integer"
                null={false}
                required={true}
                type="IntegerField"
                upperFirstLabel={true}
                verbose_name="integer"
              />,
              <IntegerField
                blank={false}
                default={0}
                displayOptions={
                  Object {
                    "width": 16,
                  }
                }
                editable={true}
                formId="bla"
                help_text=""
                name="f_positive_integer"
                null={false}
                required={true}
                type="PositiveIntegerField"
                upperFirstLabel={true}
                verbose_name="positive integer"
              />,
              <IntegerField
                blank={false}
                default={13}
                displayOptions={
                  Object {
                    "width": 16,
                  }
                }
                editable={true}
                formId="bla"
                help_text=""
                name="f_small_integer"
                null={false}
                required={true}
                type="SmallIntegerField"
                upperFirstLabel={true}
                verbose_name="small integer"
              />,
              <IntegerField
                blank={false}
                default={222}
                displayOptions={
                  Object {
                    "width": 16,
                  }
                }
                editable={true}
                formId="bla"
                help_text=""
                name="f_positive_small_integer"
                null={false}
                required={true}
                type="PositiveSmallIntegerField"
                upperFirstLabel={true}
                verbose_name="positive small integer"
              />,
              <IntegerField
                blank={false}
                decimal_places={4}
                default={3.1415}
                displayOptions={
                  Object {
                    "width": 16,
                  }
                }
                editable={true}
                formId="bla"
                help_text=""
                max_digits={10}
                name="f_decimal"
                null={false}
                required={true}
                type="DecimalField"
                upperFirstLabel={true}
                verbose_name="decimal"
              />,
              <IntegerField
                blank={false}
                default={3.14}
                displayOptions={
                  Object {
                    "width": 16,
                  }
                }
                editable={true}
                formId="bla"
                help_text=""
                name="f_float"
                null={false}
                required={true}
                type="FloatField"
                upperFirstLabel={true}
                verbose_name="float"
              />,
              <InlineForeignKeyField
                blank={true}
                data={
                  Array [
                    Object {
                      "id": 1,
                      "name": "Author Вася",
                    },
                    Object {
                      "id": 2,
                      "name": "Author asd Вася",
                    },
                  ]
                }
                displayOptions={
                  Object {
                    "width": 16,
                  }
                }
                editable={true}
                formId="bla"
                getOptionLabel={[Function]}
                getOptionValue={[Function]}
                help_text=""
                isRtl={false}
                name="f_fk_embed"
                null={true}
                required={false}
                type="ForeignKey"
                upperFirstLabel={true}
                verbose_name="embedded foreign key"
              />,
              <GenericField
                blank={true}
                data="http://localhost:8000/api/v1/sample/books/"
                displayOptions={
                  Object {
                    "width": 16,
                  }
                }
                editable={true}
                formId="bla"
                help_text=""
                name="f_fk_rel"
                null={true}
                required={false}
                type="ForeignKey"
                upperFirstLabel={true}
                verbose_name="foreign key with relation"
              />,
              <AutocompleteChoiceField
                blank={false}
                choices={
                  Array [
                    Array [
                      0,
                      "dead",
                    ],
                    Array [
                      255,
                      "alive",
                    ],
                    Array [
                      30,
                      "dried",
                    ],
                  ]
                }
                default={255}
                displayOptions={
                  Object {
                    "width": 16,
                  }
                }
                editable={true}
                formId="bla"
                getOptionLabel={[Function]}
                getOptionValue={[Function]}
                help_text=""
                name="f_choice"
                null={false}
                required={true}
                type="PositiveSmallIntegerField"
                upperFirstLabel={true}
                verbose_name="choice"
              />,
              <GenericField
                blank={false}
                displayOptions={
                  Object {
                    "width": 16,
                  }
                }
                editable={true}
                formId="bla"
                help_text=""
                max_length={100}
                name="f_file"
                null={false}
                required={true}
                type="FileField"
                upload_url="http://localhost:8000/api/v1/sample/all/accept_file/"
                upperFirstLabel={true}
                verbose_name="short preview sample"
              />,
              <GenericField
                blank={true}
                displayOptions={
                  Object {
                    "width": 16,
                  }
                }
                editable={true}
                formId="bla"
                help_text="logo image"
                max_length={100}
                name="f_photo"
                null={true}
                required={false}
                type="FileField"
                upload_url="http://localhost:8000/api/v1/sample/all/accept_file/"
                upperFirstLabel={true}
                verbose_name="photo"
              />,
              <CharField
                blank={false}
                displayOptions={
                  Object {
                    "width": 16,
                  }
                }
                editable={true}
                formId="bla"
                help_text="charfield"
                max_length={255}
                name="f_char_field"
                null={false}
                required={true}
                type="CharField"
                upperFirstLabel={true}
                verbose_name="char field"
              />,
              <TextField
                blank={true}
                displayOptions={
                  Object {
                    "width": 16,
                  }
                }
                editable={true}
                formId="bla"
                help_text=""
                name="f_text_field"
                null={false}
                required={false}
                type="TextField"
                upperFirstLabel={true}
                verbose_name="text field"
              />,
              <InlineManyToManyField
                blank={true}
                closeOnSelect={true}
                data={
                  Array [
                    Object {
                      "id": 1,
                      "name": "Author Вася",
                    },
                    Object {
                      "id": 2,
                      "name": "Author asd Вася",
                    },
                  ]
                }
                displayOptions={
                  Object {
                    "width": 16,
                  }
                }
                editable={true}
                formId="bla"
                getOptionLabel={[Function]}
                getOptionValue={[Function]}
                help_text=""
                name="f_m2m_embed"
                null={false}
                required={false}
                type="ManyToManyField"
                upperFirstLabel={true}
                value={Array []}
                verbose_name="embedded M2M"
              />,
              <GenericField
                blank={true}
                data="http://localhost:8000/api/v1/sample/authors/"
                displayOptions={
                  Object {
                    "width": 16,
                  }
                }
                editable={true}
                formId="bla"
                help_text=""
                name="f_m2m_rel"
                null={false}
                required={false}
                type="ManyToManyField"
                upperFirstLabel={true}
                verbose_name="related M2M"
              />,
            ],
            "header": null,
          },
        ],
        "formId": undefined,
        "isTitleVisible": true,
        "loading": false,
        "nonFieldErrorsMap": undefined,
        "onSubmit": [Function],
        "submitComponent": [Function],
        "title": "Create AllModelFields",
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
